### PR TITLE
ppc64 port

### LINKED ISSUE #5: Duplicate Target "roco2_ariel"
Both [`src/configurations/ariel/CMakeLists.txt`](https://github.com/tud-zih-energy/roco2/blob/master/src/configurations/ariel/CMakeLists.txt) and [`src/configurations/conway/CMakeLists.txt`](https://github.com/tud-zih-energy/roco2/blob/master/src/configurations/conway/CMakeLists.txt) invoke a macro to create a target `roco2_ariel` which is thus a duplicate and causes cmake to crash at configuration time. (See below for a full log)

I suggest changing the name or disabling building the examples by default.

```
-- Checking git submodules in /tmp/tmp.AlcMpemkEl/roco2
-- SCOREP library found. (using /opt/scorep/bin/scorep-config)
-- Building WITHOUT Score-P user instrumentation.
-- Could NOT find X86Adapt (missing: X86_ADAPT_LIBRARIES X86_ADAPT_INCLUDE_DIRS)
-- X86Adapt wasn't found, DDCM not available.
CMake Error at CMakeLists.txt:74 (add_executable):
  add_executable cannot create target "roco2_ariel" because another target
  with the same name already exists.  The existing target is an executable
  created in source directory
  "/tmp/tmp.AlcMpemkEl/roco2/src/configurations/ariel".  See documentation
  for policy CMP0002 for more details.
Call Stack (most recent call first):
  src/configurations/conway/CMakeLists.txt:1 (Roco2Configuration)


-- Configuring incomplete, errors occurred!
See also "/tmp/tmp.AlcMpemkEl/roco2/build/CMakeFiles/CMakeOutput.log".
See also "/tmp/tmp.AlcMpemkEl/roco2/build/CMakeFiles/CMakeError.log".
```

### LINKED ISSUE #7: port to ppc64
normal build process should work on ppc64 machines, will test on ppc64le

concept: use cmake magic to detect arch and disable incompatible source parts

### PR DESCRIPTION
**Pull request by s9105947**
Closes #7.

Also adds two new configurations, see `src/configurations/p9*`.

Also fixes #5 and #8.

### CODE REVIEW

File: include/roco2/kernels/high_low.hpp [hunk 1]
```diff
@@ -20,6 +23,18 @@ namespace kernels
         {
         }
 
+        roco2::chrono::time_point::duration get_high_time() const {
```
Comment (bmario) [inline include/roco2/kernels/high_low.hpp:26]: we are following STL-alike coding conventions in this project, so I'd prefer getter without the `get_` prefix, like `tag()`.

File: include/roco2/kernels/high_low.hpp [hunk 2]
```diff
@@ -32,25 +47,41 @@ namespace kernels
             SCOREP_USER_REGION("high_low_bs_kernel", SCOREP_USER_REGION_TYPE_FUNCTION)
 #endif
             roco2::chrono::time_point deadline = std::chrono::high_resolution_clock::now();
+            // align across all procs
+            deadline -= deadline.time_since_epoch() % get_period();
+
+            // record highlow frequency to trace
+            roco2::metrics::frequency::instance().write(std::chrono::seconds(1) / std::chrono::duration_cast<std::chrono::duration<double>>(get_period()));
```
Comment (bmario) [inline include/roco2/kernels/high_low.hpp:54]: The frequency metric is used to write the actual core/processor frequency. I think we could use the `utilization` metric instead.

File: CMakeLists.txt [hunk 3]
```diff
@@ -23,28 +26,41 @@ SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=c99")
...
+    message(STATUS "silencing all logging")
+endif()
+
 find_package(OpenMP REQUIRED)
 target_link_libraries(roco2 INTERFACE OpenMP::OpenMP_CXX)
 
 # this should also find MKL
 find_package(BLAS REQUIRED)
-target_link_libraries(roco2 INTERFACE BLAS::BLAS)
+target_link_libraries(roco2 INTERFACE ${BLAS_LIBRARIES})
```
Comment (bmario) [inline CMakeLists.txt:42]: I think I'd rather increase the required CMake version to 3.18.

**Review by bmario — CHANGES_REQUESTED** [review]

File: include/roco2/kernels/high_low.hpp (hunk 1 above)
Comment (s9105947) [reply include/roco2/kernels/high_low.hpp]: implemented in 3e199f6

File: include/roco2/kernels/high_low.hpp (hunk 2 above)
Comment (s9105947) [reply include/roco2/kernels/high_low.hpp]: implemented in 3e199f6

File: CMakeLists.txt (hunk 3 above)
Comment (s9105947) [reply CMakeLists.txt:42]: I'd argue against increasing the required cmake version:

1. this is the only part in the code where a higher cmake version would be required
2. research systems are riddled with old releases (including our local testing systems, a recompilation of at least 2 libs would be required) -- why break compatibility (jumping 7 minor versions!) for an optical change

**Pull request merged**